itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java

```diff
@@ -263,6 +266,21 @@ WarehouseInstance runFailure(String command, int errorCode) throws Throwable {
     }
   }
 
+  WarehouseInstance verifyNotificationID(String dumpLocation) throws Exception {
+    FileSystem fs = new Path(dumpLocation).getFileSystem(hiveConf);
+    long currentNotificationID = getCurrentNotificationEventId().getEventId();
+    Path notifyFilePath = new Path(dumpLocation + "/hive/", "_notification_id");
+    assertTrue(fs.exists(notifyFilePath));
+    InputStream inputStream = fs.open(notifyFilePath);
+
+    BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+    for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+      assertTrue(currentNotificationID > Long.parseLong(line));
```

Comment:
This may pass for 0 as well. What we need to make sure is that we are making progress on each iteration of dump-load.
Store the currently stored notification id in a variable. And perform an additional assertion that the current one is greater than previously stored one.